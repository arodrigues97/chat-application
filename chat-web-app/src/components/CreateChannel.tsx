import { ChangeEvent } from "react"
import { Button, Form, Header, Icon, Modal, Segment } from "semantic-ui-react"

export type CreateChannelProps = {
  channelName: string
  handleChannelNameChange: (event: ChangeEvent<HTMLInputElement>) => void
  handleCreateChannel: () => void
}

const CreateChannel = (props: CreateChannelProps) => {
  const { channelName, handleChannelNameChange, handleCreateChannel } = props

  return (
    <Segment placeholder>
      <Header icon>
        <Icon name="question" />
        Create your own channel
      </Header>
      <Modal
        trigger={<Button icon name="plus" primary content="Create Channel" />}
        header="Create a Channel"
        content={
          <Modal.Content>
            <Form>
              <Form.Field>
                <Form.Input
                  value={channelName}
                  onChange={handleChannelNameChange}
                  type="text"
                  placeholder="Enter the channel name"
                />
              </Form.Field>
            </Form>
          </Modal.Content>
        }
        actions={[
          "Cancel",
          {
            key: "yes",
            content: "Create",
            positive: true,
            onClick: () => handleCreateChannel(),
          },
        ]}
      />
    </Segment>
  )
}

export default CreateChannel
