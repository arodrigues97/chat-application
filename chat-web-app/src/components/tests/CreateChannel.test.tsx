import { mount } from "enzyme"
import { ChangeEvent } from "react"
import CreateChannel from "../CreateChannel"

describe("CreateChannel", () => {
  let channelName = ""
  let channelCreated = false
  const component = (
    <CreateChannel
      channelName={channelName}
      handleChannelNameChange={function (
        event: ChangeEvent<HTMLInputElement>
      ): void {
        channelName = event.target.value
      }}
      handleCreateChannel={function (): void {
        channelCreated = true
      }}
    />
  )

  let wrapper: any

  beforeEach(() => {
    wrapper = mount(component)
  })

  it("should render initial layout", () => {
    expect(wrapper.getElement()).toMatchSnapshot()
  })

  it("renders create button", () => {
    const createButton = wrapper.find("button")
    const label = createButton.text()
    expect(label).toEqual("Create Channel")
  })

  it("create button click should open modal", () => {
    const createButton = wrapper.find("button")
    createButton.simulate("click")
    expect(wrapper.find("form").length).toBe(1)
  })

  it("handles channel name input", () => {
    const createButton = wrapper.find("button")
    createButton.simulate("click")
    const channelNameInput = wrapper.find("input").at(0)
    channelNameInput.instance().value = "Test"
    channelNameInput.simulate("change")
    expect(channelName).toBe("Test")
  })

  it("handles channel create button", () => {
    const createButton = wrapper.find("button")
    createButton.simulate("click")
    const channelNameInput = wrapper.find("input").at(0)
    channelNameInput.instance().value = "Test"
    channelNameInput.simulate("change")
    wrapper.find("button").at(1).simulate("click")
    expect(channelCreated).toBe(true)
  })
})
