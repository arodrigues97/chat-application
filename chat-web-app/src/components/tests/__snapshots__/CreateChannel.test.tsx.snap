// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateChannel should render initial layout 1`] = `
<CreateChannel
  channelName=""
  handleChannelNameChange={[Function]}
  handleCreateChannel={[Function]}
/>
`;
